SEP-0012.md

```diff
@@ -0,0 +1,553 @@
+            associated with a single pixel dimension, this WCS will only have 1 pixel dimension,
+            even if the `.NDCube` object has a data array of 2-D or greater.
+            Therefore using this WCS directly might lead to some confusing results.
+
+        """
+
+    @abc.abstractmethod
+    def __getitem__(self, item: Union[str, int, slice, Iterable[Union[int, slice]]]) -> "ExtraCoordsABC":
+        """
+        ExtraCoords can be sliced with either a string, or a numpy like slice.
+
+        When sliced with a string it should return a new ExtraCoords object
+        with only those coordinates with the given names. When sliced with a
+        numpy array like slice it should return a new ExtraCoords with the
+        slice applied. Supporting step is not required and "fancy indexing" is
+        not supported.
+        """
+```
+
+## Global Coords
+
+The final part of the `NDCube` API is a class to provide structured metadata for coordinate values which are applicable to the whole cube.
+One possible use (not mandated by this SEP) is to have any coordinates dropped by a slicing operation on a higher dimensionality cube be tracked here.
+This class is only a slight extension of the API for an immutable dictionary, to enable the tracking of the physical type of the coordinate as well as it's name and value.
+The class behaves like a mapping between coordinate name and coordinate object for all read operations.
+
+
+```python
+
+from collections.abc import Mapping
+
+
+class GlobalCoordsABC(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+
+    This class acts as a mapping between coordinate name and the coordinate object.
+    In addition to this a physical type is stored for each coordinate name.
+    A concrete implementation of this class must furfill the `Mapping` ABC,
+    including methods such as `__iter__` and `__len__`.
+
+    Parameters
+    ----------
+    ndcube : `.NDCube`, optional
+        The parent ndcube for this object. Used to extract global coordinates
+        from the wcs and extra coords of the ndcube. If not specified only
+        coordinates explicitly added will be shown.
+    """
+    def __init__(self, ndcube: Optional[NDCubeABC] = None):
+        self._ndcube = ndcube
+
+    def add(self,
+            name: str,
+            physical_type: str,
+            coord: Any
+            ):
+        """
+        Add a new coordinate to the collection.
+
+        Parameters
+        ----------
+        name
+            The name for the coordinate.
+        physical_type
+            An IOVA UCD1+ physical type description for the coordinate
+            (http://www.ivoa.net/documents/latest/UCDlist.html). If no matching UCD
+            type exists, this can instead be ``"custom:xxx"``, where ``xxx`` is an
+            arbitrary string. If not known, can be `None`.
```

Comment:
Not sure what are the options here: String `"custom:None"`, `"None"`, or `None`?

Reply:
The options are one of the valid UCD1+ ids, any string prefixed with `"custom:"` or the `None` type if no physical type information is known.